packages/compass-telemetry/src/hooks.ts

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import type { TrackFunction } from './track';
+import { useTelemetry } from './provider';
+
+/**
+ * Hook that allows to track telemetry events as a side effect of dependencies changing.
+ *
+ * @param {function(TrackFunction): void} onChange - Function to be called when dependencies change. Receives the current track as an argument.
+ * @param {unknown[]} dependencies - Array of dependencies to watch for changes.
+ * @param {Object} [options]
+ * @param {boolean} [options.skipOnMount=false] - If true, the onChange function is skipped on the initial mount.
+ *
+ * @example
+ * useTrackOnChange((track) => {
+ *   if (isShellOpen) { track('Shell Opened') }
+ * }, [isShellOpen], { skipOnMount: true });
+ */
+export function useTrackOnChange(
+  onChange: (track: TrackFunction) => void,
+  dependencies: unknown[],
+  options: { skipOnMount: boolean } = { skipOnMount: false }
+) {
+  const onChangeRef = React.useRef(onChange);
+  onChangeRef.current = onChange;
+  const track = useTelemetry();
+  let initial = true;
+  React.useEffect(() => {
+    if (options.skipOnMount && initial) {
```

Comment:
Is this a pre-existing bug? `initial` is never `false` here, is it? Should it be created via `useState()`?